packages/svelte/src/compiler/compile/Component.js

```diff
@@ -772,12 +772,23 @@ export default class Component {
 			}
 			const writable =
 				node.type === 'VariableDeclaration' && (node.kind === 'var' || node.kind === 'let');
-			const imported = node.type.startsWith('Import');
+
+			let immutable = false;
+			if (node.type === 'VariableDeclaration' && node.kind === 'const') {
+				immutable = true;
+				for (const declaration of node.declarations) {
+					if (declaration.init.type !== 'Literal' || typeof declaration.init.value === 'object') {
+						immutable = false;
+					}
+				}
+			}
+
 			this.add_var(node, {
 				name,
 				initialised: instance_scope.initialised_declarations.has(name),
+				imported: node.type.startsWith('Import'),
 				writable,
-				imported
+				immutable
```

Comment:
What's the difference of setting immutable, vs checking if the variable has been reassigned / mutated?

Reply:
I suppose for this PR we could just check if the variable has been reassigned or mutated. What I really want to do is handle things like `import MyImg from './myimg.png'` as immutable and this PR was kind of a warm up for that. (This is my first real PR to the compiler, so still learning!) I think with the `import` case we can't easily check if it's been mutated because it possibly could be mutated in another file and so I was planning to extend the logic for `immutable` and add other cases to it.

Reply:
hmm.. although imported variable is a live binding and can be mutated from another file, i dont think svelte currently tracks / able to treat imports as reactive variables

Reply:
ooh, yeah, you're right about that. I updated it to use that logic, which is great as it accomplishes my end goal